### PR TITLE
Some Tweaks To the Menu for better UX and Extendibility

### PR DESCRIPTION
**Pull request by quickfingerz**
Just a few simple edits that I needed to make to allow some menu extensions like sliders and such, being able to override UpdateText() was necessary for some nicer menu components.

### CODE REVIEW

File: source/MenuItem.hpp [hunk 1]
```diff
@@ -57,17 +57,17 @@ namespace GTA
 		virtual void Change(bool right);
 		virtual void SetOriginAndSize(System::Drawing::Point topLeftOrigin, System::Drawing::Size size);
 
-		void UpdateText();
+		virtual void UpdateText();
 
 		virtual property MenuBase ^Parent;
 		virtual property System::String ^Caption;
 		virtual property System::String ^Description;
+		property UIText ^mText;
```
Comment (JohnnyCrazy) [inline source/MenuItem.hpp]: Any reason for this property? Or did you implement it so there is more freedom for devs?

Comment (quickfingerz) [inline source/MenuItem.hpp]: I see what you mean, the reason is the result we want is a virtual UpdateText as it does bespoke formatting per component, (such as adding the numbers and < > signs at the end of the NumericScrollers and the like). As it speaks directly to the mText property I just made it public for now

**Comment by JohnnyCrazy** [comment]
Since we merged #121, there is no actuall need for an `UpdateText`-Method. Setting the property `Caption` will now update the text accordingly.

But we should definitely merge `TextOffset` + the ability to jump back to the first/last element when reaching one end of the menu.

@quickfingerz Could you make a commit reversing the UpdateText + property stuff? That would be awsome :+1:

**Comment by quickfingerz** [comment]
I merged in the current changes from the master, and that reverted the update text stuff back to the way you have it, and kept the wrapping and text offset, hopefully that should do it! I think I accidentally added the commit which changed the reload key, that was a personal preference, sorry bout that, I'm new to this pull request thing

**Comment by Frede1012** [comment]
@quickfingerz You should probably undo the reload key change, but otherwise it looks good! Maybe UpdateText should still be virtual even though its private?

**Comment by quickfingerz** [comment]
and done :)

**Comment by Frede1012** [comment]
@quickfingerz What about the virtual UpdateText function?

**Comment by JohnnyCrazy** [comment]
Any reason to keep it @Frede1012 ?

**Comment by Frede1012** [comment]
@JohnnyCrazy For overriding in case you want to create your own MenuItems in C#.

**Comment by JohnnyCrazy** [comment]
What if you have MenuItems that don't have any text at all?

Just create a method on your own
````c#
public void UpdateText()
{
    this.Caption = "[CUSTOM]" + this.Caption;
}
````

**Comment by quickfingerz** [comment]
its straight forward enough to derive from menu item in c# and all is well, update text is just used on some of the current menu items but I agree with Johnnycrazy that it doesn't need to be in the interface or virtual

**Comment by Frede1012** [comment]
Nevermind then.

**Pull request merged**